server/API/ListLectures.php

```diff
@@ -0,0 +1,270 @@
+			}
+
+			$userType = intval($userData['type']);
+
+			switch ($userType) {
+				case USER_TYPE_STUDENT:
+					$stmt = $pdo->prepare($studentQuery);
+					$stmt->bindValue(':online', LECTURE_REMOTE, PDO::PARAM_INT);
+					$stmt->bindValue(':studentId', $userId, PDO::PARAM_INT);
+					$stmt->bindValue(':cancelled', LECTURE_CANCELLED, PDO::PARAM_INT);
+					$stmt = bindTimeConstraints(
+						$stmt,
+						isset($_GET['startDate']) ? $_GET['startDate'] : null,
+						isset($_GET['endDate']) ? $_GET['endDate'] : null
+					);
+					break;
+				case USER_TYPE_TEACHER:
+					$stmt = $pdo->prepare($teacherQuery);
+					$stmt->bindValue(':online', LECTURE_REMOTE, PDO::PARAM_INT);
+					$stmt->bindValue(':teacherId', $userId, PDO::PARAM_INT);
+					$stmt->bindValue(':cancelled', LECTURE_CANCELLED, PDO::PARAM_INT);
+					$stmt = bindTimeConstraints(
+						$stmt,
+						isset($_GET['startDate']) ? $_GET['startDate'] : null,
+						isset($_GET['endDate']) ? $_GET['endDate'] : null
+					);
+					break;
+				case USER_TYPE_BOOK_MNGR:
+					$stmt = $pdo->prepare($bookMngrQuery);
+					$stmt->bindValue(':online', LECTURE_REMOTE, PDO::PARAM_INT);
+					$stmt->bindValue(':cancelled', LECTURE_CANCELLED, PDO::PARAM_INT);
+					$stmt = bindTimeConstraints(
+						$stmt,
+						isset($_GET['startDate']) ? $_GET['startDate'] : null,
+						isset($_GET['endDate']) ? $_GET['endDate'] : null
+					);
+					break;
+				default:
+					throw new Exception("Unknown user type " . $userType);
+					break;
+			}
+
+			if (!$stmt->execute()) {
+				throw new PDOException($stmt->errorInfo()[2]);
+			}
+
+			$lectures = $stmt->fetchAll(PDO::FETCH_ASSOC);
+
+			echo json_encode([
+				'success' => true,
+				'lectures' => array_map(
+					function ($e) {
+						return array(
+							'lectureId' => intval($e["lectureId"]),
+							'courseId' => intval($e["courseId"]),
+							'courseName' => $e["courseName"],
+							'startTS' => intval($e["startTS"]),
+							'endTS' => intval($e["endTS"]),
+							'online' => boolval($e["online"]),
+							'teacherName' => $e["teacherName"],
+							'roomName' => $e["roomName"],
+							'bookedSeats' => intval($e["bookedSeats"]),
+							'totalSeats' => intval($e["totalSeats"]),
+							'bookedSelf' => isset($e["bookedSelf"]) ? boolval($e["bookedSelf"]) : null,
+							'inWaitingList' => isset($e["inWaitingList"]) ? boolval($e["inWaitingList"]) : null,
+							'waitingUsers' => isset($e["waitingUsers"]) ? intval($e["waitingUsers"]) : null,
+							'cancelledBookings' => isset($e["cancelledBookings"]) ? boolval($e["cancelledBookings"]) : null,
+						);
+					},
+					$lectures
+				)
+			]);
+		} catch (Exception $e) {
+			echo json_encode([
+				'success' => false,
+				'reason' => $e->getMessage()
+			]);
+		}
+	}
+}
+
+
+if (!function_exists('bindTimeConstraints')) {
+
+
+	function bindTimeConstraints($pdoStmt, $startISODate, $endISODate) {
+		$startTS = 0;
+		$endTS = 9e9;
+
+		if ($startISODate) {
+			$Ymd = explode('-', $startISODate);
```

Comment:
why don't you use DateTime? Like, for example $Ymd = new DateTime($startISODate); $startTS = $Ymd->getTimestamp;